x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/application/analytics/ingest/BulkProcessorFactory.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+package org.elasticsearch.xpack.application.analytics.ingest;
+
+import org.elasticsearch.action.bulk.BulkItemResponse;
+import org.elasticsearch.action.bulk.BulkProcessor2;
+import org.elasticsearch.action.bulk.BulkRequest;
+import org.elasticsearch.action.bulk.BulkResponse;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.common.inject.Inject;
+import org.elasticsearch.common.unit.ByteSizeUnit;
+import org.elasticsearch.common.unit.ByteSizeValue;
+import org.elasticsearch.logging.LogManager;
+import org.elasticsearch.logging.Logger;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Event ingest is done through a {@link BulkProcessor2}. This class is responsible for instantiating the bulk processor.
+ */
+public class BulkProcessorFactory {
+    private static final Logger logger = LogManager.getLogger(AnalyticsEventEmitter.class);
+
+    private final BulkProcessorConfig config;
+
+    @Inject
+    public BulkProcessorFactory(BulkProcessorConfig config) {
+        this.config = config;
+    }
+
+    public BulkProcessor2 create(Client client) {
+        return BulkProcessor2.builder(client::bulk, new BulkProcessorListener(), client.threadPool())
```

Comment:
I will probably implement it into the `AnalyticsEventEmitter`.
If you don't mind I will do it in a follow-up PR to be able to merge the PR and unblock other work streams that rely on the feature.

Reply:
yes, follow up is good. But I will this up to ent search team to decide if this is even needed at all. I thought that it might be good as an emergency switch off button 

We have that for deprecation logging, but we started with that feature turned off so we needed this to enable this.

Reply:
This is a good call and I will add these feature. It will probably be part of the upcoming PR to add metering to the event ingest API.